packages/babel-traverse/src/scope/index.js

```diff
@@ -90,7 +90,9 @@ const collectorVisitor = {
     //if (path.isFlow()) return;
 
     // we've ran into a declaration!
-    path.scope.getFunctionParent().registerDeclaration(path);
+    const parent =
+      path.scope.getFunctionParent() || path.scope.getProgramParent();
```

Comment:
@nicolo-ribaudo you're correct, that's why I made it to return `null` if it can't find a function path - https://github.com/babel/babel/pull/5923#discussion_r127225252